mlte/frontend/nuxt-app/composables/artifact-validation.ts

```diff
@@ -0,0 +1,44 @@
+// artifact-validation.ts - provides functions to validate that JSON artifacts comply with their schemas.
+
+import { Validator } from 'jsonschema';
+
+import * as negotiationSchemaData from '~/assets/schema/artifact/spec/v0.0.1/schema.json';
+import * as specSchemaData from '~/assets/schema/artifact/spec/v0.0.1/schema.json';
+import * as validatedSchemaData from '~/assets/schema/artifact/spec/v0.0.1/schema.json';
+import * as valueSchemaData from '~/assets/schema/artifact/spec/v0.0.1/schema.json';
+
+
+export function isValidNegotiation(artifact: object) : boolean {
+    return isValidArtifact(artifact, negotiationSchemaData);
+}
+
+export function isValidSpec(artifact: object) : boolean {
+    return isValidArtifact(artifact, specSchemaData);
+}
+
+export function isValidValidatedSpec(artifact: object) : boolean {
+    return isValidArtifact(artifact, validatedSchemaData);
+}
+
+export function isValidValue(artifact: object) : boolean {
+    return isValidArtifact(artifact, valueSchemaData);
+}
+
+// Validates if a given JSON string is a valid JSON artifact for the provided schema.
+function isValidArtifact(artifact: object, schema: object) : boolean {
+    if(!("body" in artifact)) {
+        console.log("INVALID!!! Object has no body section.")
```

Comment:
Some of these printouts were left in for now to help debugging when integrating, but they can be removed later.

Reply:
Good call.